src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,342 @@
+            ).model_dump(mode="json"),
+        )
+        execution_node = ExecutionNode(request_task)
+
+        # Get the custom request fields from the privacy request.
+        # Returns something like {"site_id": {"value": "1234", "label": "Site Id"}}
+        custom_request_fields = (
+            privacy_request.get_persisted_custom_privacy_request_fields()
+        )
+
+        # We execute a standalone retrieval query on the connector to get the email address
+        # based on the custom request fields on the privacy request and the field_name
+        # provided as part of the connection config.
+        # Following the previous example, this query would be something like
+        # SELECT email_address FROM site_info WHERE site_id = '1234'
+        email_recipient_data = connector.execute_standalone_retrieval_query(
+            node=execution_node,
+            fields=[field_name],
+            filters={
+                custom_field_name: [custom_field_data["value"]]
+                for custom_field_name, custom_field_data in custom_request_fields.items()
+            },
+        )
+        if not email_recipient_data:
+            logger.error(
+                "Custom request field lookup yielded no results. Skipping email send for privacy request with id {}.",
+                privacy_request.id,
+            )
+            return None
+
+        if len(email_recipient_data) > 1:
+            logger.error(
+                "Custom request field lookup yielded multiple results. Skipping email send for privacy request with id {}.",
+                privacy_request.id,
+            )
+            return None
+
+        return email_recipient_data[0][field_name]
+
+    def batch_email_send(self, privacy_requests: Query) -> None:
+        # Import here to avoid circular import error
+        # TODO: once we incorporate custom request fields into the DSR graph logic
+        # we won't need this import here, so ignorning cyclic import warning for now
+        from fides.api.service.connectors import (  # pylint: disable=cyclic-import
+            get_connector,
+        )
+
+        logger.debug(
+            "Starting batch_email_send for connector: {} ...", self.configuration.name
+        )
+
+        db: Session = Session.object_session(self.configuration)
+
+        dataset_reference = self.config.recipient_email_address
+        dataset_key = dataset_reference.dataset  # pylint: disable=no-member
+
+        # We get the DatasetConfig instance using the dataset key provided in the configuration
+        dataset_config: Optional[DatasetConfig] = DatasetConfig.get_by(
+            db, field="fides_key", value=dataset_key
+        )
+
+        if not dataset_config:
+            logger.error(
+                "DatasetConfig with key '{}' not found. Skipping erasure email send for connector: '{}'.",
+                dataset_key,
+                self.configuration.key,
+            )
+            return
```

Comment:
I would raise an exception here just in case so the customer sees the failed privacy requests and we can work on addressing the connector instead of having this issue go undetected

Reply:
I don't think raising the exception is enough for the error to be visible , I think I also need to add an ExecutionLog for each privacy request. I'll do that as well to ensure the privacy request shows as failed. Also if I raise the exception it seems like the task keeps retrying which is probably not what we want?

Reply:
ah right we're outside of that typical flow